### ISSUE TITLE
Bug: [no-confusing-non-null-assertion] should not report if nonnull and operator are on separate lines

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#9995]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3lAXPTMAligObwA%2B8KArhBADTwBGeBxJA3AFBeEBm8ABRd42AIQj4AXimSmXAJTwA3gF8OQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1rI6YDNYyZgHNanJmITxaAQ2TJE0fJV7ooAdxnQmkcGAC%2BIfUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const a: string | null, b: string;

if (
  a!
  ==
  b
) {};
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-confusing-non-null-assertion": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expect that no report should occur, because this is not really confusing.

### Actual Result

The code is flagged for possibly looking like `!==`

**Comment by bradzacher** [comment]
I'd personally disagree I don't think that you should be able to hack around the rule by adding a newline.

Also this style of formatting is... Uncommon - so I would also disagree that it's worth us thinking and adding handling for this.

**Comment by kirkwaiblinger** [comment]
Fair, I don't feel strongly 👍
